src/lsp_data.rs

```diff
@@ -395,3 +416,30 @@ pub struct ProgressParams {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub done: Option<bool>,
 }
+
+pub fn transform_json_key_one_level(mut val: serde_json::Value) -> serde_json::Value {
```

Comment:
For clarity's sake, could we just use the single `fn (mut Value) -> Value` function variant?

Also, could we try and come up with a more descriptive name? I couldn't understand at first what the transformation this actually does.